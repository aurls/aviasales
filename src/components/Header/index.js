import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './header.scss';
import logo from './logo.png';

const Header = ({ appearance }) => {
  let logoStyle = 'header__logo';
  if (appearance.isFetching) logoStyle += ' is-fetching';

  return (
    <header className="header">
      <img className={logoStyle} src={logo} alt="logo" />
    </header>
  );
};

const mapStateToProps = (state) => {
  return {
    appearance: state.appearance
  };
};

Header.propTypes = {
  appearance: PropTypes.object.isRequired
};

export default connect(mapStateToProps)(Header);
