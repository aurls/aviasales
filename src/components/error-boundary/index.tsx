import React from 'react';
import ErrorMessage from '../error-message';
import './error-boundary.scss';

type Props = {
  children: React.ReactNode
}

type State = {
  hasReactError: boolean
}

class ErrorBoundary extends React.Component<Props, State> {
  state: State = {
    hasReactError: false
  };

  componentDidCatch () {
    this.setState({
      hasReactError: true
    });
  }

  render () {
    const { hasReactError } = this.state;
    if (hasReactError) {
      return (
        <div className="error-boundary">
          <ErrorMessage />
        </div>);
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
