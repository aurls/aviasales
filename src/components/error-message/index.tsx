import React from 'react';
import './error-message.scss';

const ErrorMessage: React.FC = () => {
  return (
    <section className="error-message">
      <h2 className="error-message__title">
        Ой, у&nbsp;нас что-то случилось!
      </h2>
      <p className="error-message__text">
        Мы уже знаем о&nbsp;проблеме и&nbsp;кинули все силы на&nbsp;её&nbsp;устранение.
        Попробуйте поискать через 10&nbsp;минут
      </p>
    </section>
  );
};

export default ErrorMessage;
