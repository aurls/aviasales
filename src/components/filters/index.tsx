import React from 'react';
import Types from '../../types';
import actions from '../../store/actions';
import filters from '../../services/filters';
import './filters.scss';

type Props = {
  currentFilters: Types.Filter[]
  setFilter: typeof actions.setFilter
}

const Filters: React.FC<Props> = (props: Props) => {
  const {
    currentFilters,
    setFilter
  } = props;

  const isFilterChecked = (filter: Types.Filter) => {
    return currentFilters.includes(filter) || currentFilters.includes(filters.ALL);
  };

  const onFilterClick = (filter: Types.Filter) => {
    setFilter(filter);
  };

  return (
    <div className="filters">
      <h2 className="filters__title">Количество пересадок</h2>
      <ul>
        {
          Object.values(filters).map((filter) =>
            <li key={filter.type}>
              <label className="filter">
                <input className="filter__checkbox"
                  type="checkbox"
                  name="filter"
                  value={filter.type}
                  checked={isFilterChecked(filter)}
                  onChange={() => onFilterClick(filter)} />
                {filter.title}
              </label>
            </li>)
        }
      </ul>
    </div>
  );
};

export default React.memo(Filters);
