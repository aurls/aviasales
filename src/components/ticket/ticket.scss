@import '../../assets/styles/styleConstants';

.ticket {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  margin-bottom: 20px;
  box-shadow: 0 2px 8px rgba(0, 0, 0, .1);
  border-radius: 5px;
  padding: 20px;
  background: white;
}

.ticket__price {
  width: calc(100% / 3 * 2);
  margin-top: 6px;
  font-size: 24px;
  line-height: 24px;
  font-weight: 600;
  color: $primaryColor;
}

.ticket__carrier {
  width: calc(100% / 3);
  margin-left: auto;
  margin-bottom: 20px;
  height: 36px;
}

.ticket__carrier-logo {
  height: 100%;
}

.ticket__segment {
  display: flex;
  margin-bottom: 10px;
  width: 100%;

  &:last-child {
    margin-bottom: 0;
  }
}

.ticket__time,
.ticket__duration,
.ticket__transfers {
  width: calc(100% / 3);
}

.ticket__segment-subtitle {
  font-size: 12px;
  line-height: 18px;
  font-weight: 600;
  letter-spacing: .5px;
  text-transform: uppercase;
  color: #a0b0b9;
}

.ticket__segment-info {
  font-size: 14px;
  line-height: 21px;
  font-weight: 600;
}

@media (max-width: $mobileBreakpoint) {
  .ticket__price {
    width: calc(100% / 2);
  }

  .ticket__carrier {
    width: auto;
  }

  .ticket__carrier-logo {
    margin-left: auto;
  }

  .ticket__segment {
    display: flex;
    flex-direction: column;
    width: calc(100% / 2 - 10px);
    margin-bottom: 0;
  }

  .ticket__time,
  .ticket__duration,
  .ticket__transfers {
    width: 100%;
    margin-top: 10px;
  }
}
