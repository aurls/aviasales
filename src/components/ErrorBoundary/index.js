import React from 'react';
import PropTypes from 'prop-types';

import ErrorMessage from '../ErrorMessage';
import './error-boundary.scss';

class ErrorBoundary extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      hasReactError: false
    };
  }

  componentDidCatch () {
    this.setState({
      hasReactError: true
    });
  }

  render () {
    const { hasReactError } = this.state;
    if (hasReactError) {
      return (
        <div className="error-boundary">
          <ErrorMessage />
        </div>);
    }
    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.element.isRequired
};

export default ErrorBoundary;
