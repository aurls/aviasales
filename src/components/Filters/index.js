import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import * as filters from '../../constants/filters';
import { filters as filtersActions } from '../../actions';
import './filters.scss';

const Filters = ({ appliedFilters, setFilter }) => {
  const isFilterChecked = (filter) => {
    return appliedFilters.includes(filters.ALL.type) || appliedFilters.includes(filter.type);
  };

  const onChangeFilter = (event) => {
    setFilter(event.target.value);
  };

  return (
    <div className="filters">
      <h2 className="filters__title">Количество пересадок</h2>
      <ul>
        {
          Object.values(filters).map((filter) =>
            <li key={filter.type}>
              <label className="filter"
                htmlFor={`filter-${filter.type}`}>
                <input className="filter__checkbox"
                  id={`filter-${filter.type}`}
                  type="checkbox"
                  name="filter"
                  value={filter.type}
                  checked={isFilterChecked(filter)}
                  onChange={onChangeFilter} />
                {filter.name}
              </label>
            </li>)
        }
      </ul>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    appliedFilters: state.appliedFilters
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setFilter: bindActionCreators(filtersActions.setFilter, dispatch)
  };
};

Filters.propTypes = {
  appliedFilters: PropTypes.array.isRequired,
  setFilter: PropTypes.func.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Filters);
