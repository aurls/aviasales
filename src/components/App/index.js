import React, { Fragment } from 'react';

import Header from '../Header';
import Filters from '../Filters';
import Sorting from '../Sorting';
import Tickets from '../Tickets';
import './app.scss';

class App extends React.Component {
  render () {
    return (
      <Fragment>
        <Header />
        <div className="app">
          <aside className="app__sidebar">
            <Filters />
          </aside>
          <main className="app__main">
            <Sorting />
            <Tickets />
          </main>
        </div>
      </Fragment>
    );
  }
}

export default App;
