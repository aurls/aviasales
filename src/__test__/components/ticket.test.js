import React from 'react';
import { Ticket } from '../../components/ticket';

const ticketTest = {
  price: 24769,
  carrier: 'EY',
  segments: [
    {
      origin: 'MOW',
      destination: 'HKT',
      date: '2020-02-08T20:24:00.000Z',
      stops: [],
      duration: 797
    },
    {
      origin: 'MOW',
      destination: 'HKT',
      date: '2020-02-29T7:10:00.000Z',
      stops: [],
      duration: 775
    }
  ]
};

describe('ticket component', () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <Ticket ticket={ticketTest} />
    );
  });

  it('Default state', () => {
    expect(component).toMatchSnapshot();
  });
});
