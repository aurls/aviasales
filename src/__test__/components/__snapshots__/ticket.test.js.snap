// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket component Default state 1`] = `
<section
  className="ticketTest"
>
  <div
    className="ticket__price"
  >
    24 769
     Р
  </div>
  <div
    className="ticket__carrier"
  >
    <img
      alt="EY-logo"
      className="ticket__carrier-logo"
      src="http://pics.avs.io/99/36/EY.png"
    />
  </div>
  <div
    className="ticket__segment"
    key="undefined797"
  >
    <div
      className="ticket__time"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        MOW
         – 
        HKT
      </h3>
      <p
        className="ticket__segment-info"
      >
        1:24 AM
         – 
        2:41 PM
      </p>
    </div>
    <div
      className="ticket__duration"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        В пути
      </h3>
      <p
        className="ticket__segment-info"
      >
        13ч 17м
      </p>
    </div>
    <div
      className="ticket__transfers"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        Без пересадок
      </h3>
      <p
        className="ticket__segment-info"
      />
    </div>
  </div>
  <div
    className="ticket__segment"
    key="undefined775"
  >
    <div
      className="ticket__time"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        MOW
         – 
        HKT
      </h3>
      <p
        className="ticket__segment-info"
      >
        1:10 PM
         – 
        2:05 AM
      </p>
    </div>
    <div
      className="ticket__duration"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        В пути
      </h3>
      <p
        className="ticket__segment-info"
      >
        12ч 55м
      </p>
    </div>
    <div
      className="ticket__transfers"
    >
      <h3
        className="ticket__segment-subtitle"
      >
        Без пересадок
      </h3>
      <p
        className="ticket__segment-info"
      />
    </div>
  </div>
</section>
`;
